IdentityCore/src/webview/embeddedWebview/ios/MSIDWebviewUIController.m

```diff
@@ -0,0 +1,241 @@
+    // hijack the delegate on the webview.
+    if (_webView)
+    {
+        //_webView.UIDelegate = self;
+        return YES;
+    }
+    
+    if (!_parentController)
+    {
+        _parentController = [UIApplication msalCurrentViewController];
+    }
+    
+    if (!_parentController)
+    {
+        // Must have a parent view controller to start the authentication view
+//        ADAuthenticationError* adError =
+//        [ADAuthenticationError errorFromAuthenticationError:AD_ERROR_UI_NO_MAIN_VIEW_CONTROLLER
+//                                               protocolCode:nil
+//                                               errorDetails:AD_FAILED_NO_CONTROLLER
+//                                              correlationId:nil];
+//
+//        if (error)
+//        {
+//            *error = adError;
+//        }
+        return NO;
+    }
+    
+    UIView* rootView = [[UIView alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
+    [rootView setAutoresizesSubviews:YES];
+    [rootView setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+
+    WKWebViewConfiguration *webConfiguration = [WKWebViewConfiguration new];
+    WKWebView *webView = [[WKWebView alloc] initWithFrame:rootView.frame configuration:webConfiguration];
+    [webView setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight];
+    [webView setNavigationDelegate:self];
+
+    [rootView addSubview:webView];
+    _webView = webView;
+    webView.accessibilityIdentifier = @"MSID_SIGN_IN_WEBVIEW";
+    
+    _activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
+    [_activityIndicator setColor:[UIColor blackColor]];
+    [_activityIndicator setCenter:rootView.center];
+    [rootView addSubview:_activityIndicator];
+    
+    self.view = rootView;
+    
+    UIBarButtonItem* cancelButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel
+                                                                                  target:self
+                                                                                  action:@selector(onCancel:)];
+    self.navigationItem.leftBarButtonItem = cancelButton;
+    
+    return YES;
+}
+
+/*! set webview's delegate to nil when the view controller
+ is deallocated, or it might crash ADAL. */
+-(void)dealloc
+{
+    //[_webView setUIDelegate:nil];
+    _webView = nil;
+}
+
+#pragma mark - UIViewController Methods
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    if ( (NSUInteger)[[[UIDevice currentDevice] systemVersion] doubleValue] < 7)
+    {
+        [self.navigationController.navigationBar setTintColor:[UIColor darkGrayColor]];
+    }
+}
+
+- (void)viewDidUnload
+{
+    //DebugLog();
+    
+    [super viewDidUnload];
+}
+
+- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
```

Comment:
this method is deprecated

Reply:
removed, it won't work if an app already set the rotation preference in their plist file, which is mostly the case.